splunklogger.js

```diff
@@ -513,17 +699,35 @@ SplunkLogger.prototype.flush = function (callback) {
     // After running all, if any, middlewares send the events
     var that = this;
     utils.chain(callbacks, function(err, passedContext) {
+        // The passedContext parameter could be named context to
+        // do this automatically, but the || notation adds a bit of clarity.
+        context = passedContext || context;
+
         // Errors from any of the middleware callbacks will fall through to here
         // If the context is modified at any point the error callback will get it also
         // event if next("error"); is called w/o the context parameter!
         // This works because context inside & outside the scope of this function
         // point to the same memory block.
-        // The passedContext parameter could be named context to
-        // do this automatically, but the || notation adds a bit of clarity.
         if (err) {
-            that.error(err, passedContext || context);
+            that.error(err, context);
         }
         else {
+            // Validate the context again, right before using it
+            context = that._initializeContext(context);
+            context.requestOptions.headers["Authorization"] = "Splunk " + context.config.token;
+
+            if (isBatched) {
```

Comment:
I understand what you're doing, I just don't understand why you're doing it. Why are you making a distinction here?